Warn lastValidSlot with some terminology tweaks (#15081)
solana-labs/solana: docs/src/terminology.md

## leader schedule

> A sequence of [validator](terminology.md#validator) [public keys](terminology.md#public-key) ordered by [slots](terminology.md#slot). The cluster uses the leader schedule to determine which validator is the [leader](terminology.md#leader) at any moment in time.

Reviewer: ```suggestion
A sequence of [validator](terminology.md#validator) [public keys](terminology.md#public-key) mapped to [slots](terminology.md#slot). The cluster uses the leader schedule to determine which validator is the [leader](terminology.md#leader) at any moment in time.
```

This requirement ensures no signature malleability. Each transaction must have at least one signature for [fee account](terminology#fee-account).
Thus, the first signature in transaction can be treated as [transacton id](terminology.md#transaction-id)

## skipped slot

> A past [slot](terminology.md#slot) where its scheduled leader didn't produce a [block](terminology.md#block) due to off-line or the [fork](terminology.md#fork) containing the slot was abandoned over a better alternative at the time by the cluster consensus. Being _skipped_ means that blocks at subsequent slots omits to include the skipped slots as ancestors, possibly creating an after-the-fact short cluster stall of transaction processing, oldest `recent_blockhash` expiration, [block height](terminology#block-height) increment, etc.

Author: I know we lack `recent_blockhash` definition and some relation to tx expiration as its one of data fields. but, it's time-up. ;)

Reviewer: ```suggestion
A past [slot](terminology.md#slot) that did not produce a [block](terminology.md#block), because the leader was offline or the [fork](terminology.md#fork) containing the slot was abandoned for a better alternative by cluster consensus. A skipped slot will not appear as an ancestor for blocks at subsequent slots, increment the [block height](terminology#block-height), nor expire the oldest `recent_blockhash`.
```

> By definition, being skipped can be determined only after being older than the latest [rooted](terminology.md#root) (thus not-skipped) slot.

Reviewer: ```suggestion
Whether a slot has been skipped can only be determined when it becomes older than the latest [rooted](terminology.md#root) (thus not-skipped) slot.
```

## slot

> A numbered non-overlapping slice of logical time, ordered sequentially with roughly equal passage of real wall-clock utilizing [PoH](terminology.md#proof-of-history).

Author: hehe, sorry for self-bragging but, I like this under-20-word crisp definition for slot. Previously, I think this entry lacked this crucial contextual information, immediately talking about something leader block production... :)

Reviewer: Hmm, I actually like the previous definition. How about leaving the original sentence, and then adding the contextual detail?

Author: @CriesofCarrots Got it. Thanks for honest opinion. :) How does this look now? I still tweaked the original definition a bit.

This requirement ensures no signature malleability. Each transaction must have at least one signature for [fee account](terminology#fee-account).
Thus, the first signature in transaction can be treated as [transacton id](terminology.md#transaction-id)

## skipped slot

> A past [slot](terminology.md#slot) that did not produce a [block](terminology.md#block), because the leader was offline or the [fork](terminology.md#fork) containing the slot was abandoned for a better alternative by cluster consensus. A skipped slot will not appear as an ancestor for blocks at subsequent slots, increment the [block height](terminology#block-height), nor expire the oldest `recent_blockhash`.

Reviewer: ```suggestion
A past [slot](terminology.md#slot) that did not produce a [block](terminology.md#block), because the leader was offline or the [fork](terminology.md#fork) containing the slot was abandoned for a better alternative by cluster consensus. A skipped slot will not appear as an ancestor for blocks at subsequent slots, nor increment the [block height](terminology#block-height), nor expire the oldest `recent_blockhash`.
```

Reviewer: Sorry, missed word the first time.

Whether a slot has been skipped can only be determined when it becomes older than the latest [rooted](terminology.md#root) (thus not-skipped) slot.

## slot

> The period of time for which each [leader](terminology.md#leader) ingests transactions and produces a [block](terminology.md#block) by rotation, abiding by [leader schedule](terminology.md#leader-schedule).

Reviewer: Your addition to `leader schedule` referring to slots is very good. Imo `slot` doesn't need to circle back and refer to `leader schedule`.

Reviewer: ```suggestion
The period of time for which each [leader](terminology.md#leader) ingests transactions and produces a [block](terminology.md#block).
```

> Collectively, slots create a logical clock as non-overlapping slices of the time. And they're ordered sequentially and numbered accordingly with roughly equal passage of real wall-clock utilizing [PoH](terminology.md#proof-of-history).

Reviewer: ```suggestion
Collectively, slots create a logical clock. Slots are ordered sequentially and non-overlapping, comprising roughly equal real-world time as per [PoH](terminology.md#proof-of-history).
```
